bpf/bpf_wireguard.c

```diff
@@ -275,6 +275,28 @@ int cil_from_wireguard(struct __ctx_buff *ctx)
 	bpf_clear_meta(ctx);
 	check_and_store_ip_trace_id(ctx);
 
+	/* mark packet as decrypted by wireguard */
+	ctx->mark = MARK_MAGIC_DECRYPT;
+
+#if defined(HAVE_ENCAP) && (!defined(ENABLE_NODEPORT) || !defined(ENABLE_NODE_ENCRYPTION))
```

Comment:
The equivalent of `option.Config.TunnelingEnabled()` in the DP is `defined(TUNNEL_MODE)`.
A bit counterintuitive, but `HAVE_ENCAP` can be true even in native routing (EGW, DSR Geneve).

```suggestion
#if defined(TUNNEL_MODE) && !(defined(ENABLE_NODEPORT) && defined(ENABLE_NODE_ENCRYPTION))
```

Haven't finished the review, so can't say if there's something else that might be messing in CI.